File: App/Model/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App.Model
{
    public class Customer : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler? PropertyChanged;

        public string Name { get; set; }
        public string Phone_num { get; set; }
        public int AmountOrder { get; set; }
        public decimal TotalPaid { get; set; }
        public string Rank { get; set; }

        public Customer(string name, string Phone_num, int amountOrder, decimal totalPaid, string rank)
        {
            this.Name = name;
            this.Phone_num = Phone_num;
            this.AmountOrder = amountOrder;
            this.TotalPaid = totalPaid;
            this.Rank = rank;
        }

        public Customer() { }
    }
}

File: BE/src/models/Customer.js
const mongoose = require('mongoose');

const CustomerSchema = new mongoose.Schema({
    Name: { type: String, required: true }, // Tên khách hàng
    Phone_num: { type: String, required: true }, // Số điện thoại
    AmountOrder: { type: Number, required: true }, // Số đơn hàng đã đặt
    TotalPaid: { type: Number, required: true }, // Tổng số tiền đã thanh toán
    Rank: { type: String, required: true } // Hạng khách hàng
});

const Customer = mongoose.model('customer', CustomerSchema, 'customer');
module.exports = Customer;
